src/ui/VirtualFileRenderer/VirtualFileRenderer.spec.tsx

```diff
@@ -0,0 +1,418 @@
+import * as Sentry from '@sentry/react'
+import {
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+  within,
+} from '@testing-library/react'
+import { userEvent } from '@testing-library/user-event'
+// eslint-disable-next-line no-restricted-imports
+import { type Dictionary } from 'lodash'
+import { MemoryRouter, Route, useLocation } from 'react-router-dom'
+
+import { VirtualFileRenderer } from './VirtualFileRenderer'
+
+jest.mock('@sentry/react', () => {
+  const originalModule = jest.requireActual('@sentry/react')
+  return {
+    ...originalModule,
+    captureMessage: jest.fn(),
+  }
+})
+
+window.requestAnimationFrame = (cb) => {
+  cb(1)
+  return 1
+}
+window.cancelAnimationFrame = () => {}
+
+const scrollToMock = jest.fn()
+window.scrollTo = scrollToMock
+
+class ResizeObserverMock {
+  callback = (x: any) => null
+
+  constructor(callback: any) {
+    this.callback = callback
+  }
+
+  observe() {
+    this.callback([
+      {
+        contentRect: { width: 100 },
+        target: {
+          getAttribute: () => ({ scrollWidth: 100 }),
+        },
+      },
+    ])
+  }
+  unobserve() {
+    // do nothing
+  }
+  disconnect() {
+    // do nothing
+  }
+}
+global.window.ResizeObserver = ResizeObserverMock
+
+const code = `<Breadcrumb
+    paths={[
+    { pageName: 'owner', text: owner },
+    { pageName: 'repo', text: repo },
+    ...treePaths,
+    {..props}
+    ]}
+/>`
+
+const coverageData = {
+  0: 'H',
+  1: 'M',
+  2: 'P',
+} as unknown as Dictionary<'H' | 'M' | 'P'>
+
+let testLocation: ReturnType<typeof useLocation>
+const wrapper =
+  (initialEntry = '/'): React.FC<React.PropsWithChildren> =>
+  ({ children }) => (
+    <MemoryRouter initialEntries={[initialEntry]}>
+      <Route path="/">{children}</Route>
+      <Route
+        path="*"
+        render={({ location }) => {
+          testLocation = location
+          return null
+        }}
+      />
+    </MemoryRouter>
+  )
+
+describe('VirtualFileRenderer', () => {
+  function setup() {
+    const user = userEvent.setup()
+
+    return { user }
+  }
+
+  it('renders the text-area', () => {
+    render(
+      <VirtualFileRenderer
+        code={code}
+        coverage={coverageData}
+        fileName="tsx"
+      />,
+      {
+        wrapper: wrapper(),
+      }
+    )
+
+    const textArea = screen.getByTestId('virtual-file-renderer')
+    expect(textArea).toBeInTheDocument()
+
+    const codeBlock = within(textArea).getByText(/<Breadcrumb/)
+    expect(codeBlock).toBeInTheDocument()
+  })
+
+  it('renders code in virtualized list', () => {
```

Comment:
I would test that this renders for some unknown language too. You could supply it a bad value and check that it renders (iirc as markup)

Reply:
Will do